python/google/protobuf/internal/_parameterized.py

```diff
@@ -258,7 +263,7 @@ def _ModifyClass(class_object, testcases, naming_type):
       'Cannot add parameters to %s,'
       ' which already has parameterized methods.' % (class_object,))
   class_object._id_suffix = id_suffix = {}
-  for name, obj in class_object.__dict__.items():
+  for name, obj in class_object.__dict__.copy().items():
```

Comment:
We change the size of `class_object.__dict__` inside the for loop - there's a call to `delattr(class_object, name)` in there. That causes a `RuntimeError: dictionary changed size during iteration` on Python 3.x, though Python 2.x works fine. We actually end up re-setting the deleted attribute in the same loop iteration, which might be why things work ok with Python 2.x - I'm assuming Python 3.x has more strict rules about changing the size.

Anyway, I'll add a comment summarizing that.